src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1646,19 +1655,13 @@ private unsafe ElementCount CheckUniqueAndUnfoundElements(IEnumerable<T> other,
             return result;
         }
 
-
         /// <summary>
-        /// Internal method used for HashSetEqualityComparer. Compares set1 and set2 according 
-        /// to specified comparer.
-        /// 
-        /// Because items are hashed according to a specific equality comparer, we have to resort
-        /// to n^2 search if they're using different equality comparers.
+        /// Legacy method matching previous inconsistent equality method used via nullary CreateSetComparer method.
         /// </summary>
         /// <param name="set1"></param>
         /// <param name="set2"></param>
-        /// <param name="comparer"></param>
         /// <returns></returns>
-        internal static bool HashSetEquals(HashSet<T> set1, HashSet<T> set2, IEqualityComparer<T> comparer)
+        internal static bool LegacyHashSetEquals(HashSet<T> set1, HashSet<T> set2)
```

Comment:
It would be possible to have this method exhibit the same behaviour faster, but optimising would make it less clear that the old behaviour is maintained throughout.